ComponentKit/Core/Scope/CKComponentScopeFrame.mm

```diff
@@ -88,4 +88,40 @@ - (instancetype)initWithHandle:(CKComponentScopeHandle *)handle
   return self;
 }
 
+- (void)injectScopeHierarchy:(_CKChildScopeFrameRet)oldFrame
+{
+  Class componentClass = oldFrame.cls;
+  id identifier = oldFrame.identifier;
+  CKAssert([oldFrame.frame isKindOfClass:[CKComponentScopeFrame class]], @"Can't inject with incorrect frame class.");
+
+  auto result = _children.insert({{componentClass, identifier}, oldFrame.frame});
```

Comment:
Please review this carefully, I'm not familiar enough with the code to know it it's really safe.